Spec soft-logout per MSC1466 (#2546)
matrix-org/matrix-doc: specification/client_server_api.rst

to that device. There is therefore at most one active access token assigned to
each device at any one time.

Soft logout
~~~~~~~~~~~

> When a requests fail due to a 401 status code per above, the server can

Reviewer: ```suggestion
When a request fails due to a 401 status code per above, the server can
```

include an extra response parameter, ``soft_logout``, to indicate if the client's
persisted information can be retained. This defaults to ``false``, indicating
that the server has destroyed the session. Any persisted state held by the client,

> such as encryption keys and device information, will not be able to be reused.

Reviewer: ```suggestion
such as encryption keys and device information, must not be reused and must be discarded.
```

It's not that it's not able to be reused, because the client *could* reuse it in a new login.  But the client is supposed to drop it.

When ``soft_logout`` is true, the client can acquire a new access token by
specifying the device ID it is already using to the login API. In most cases
a ``soft_logout: true`` response indicates that the user's session has expired
on the server-side and the user simply needs to provide their credentials again.

If ``soft_logout`` is ``false``, the client will not be able to reuse the device

> information it already has - the server has destroyed the session.

Reviewer: I think one of the main goals of soft logout is that it is a flag for whether the client should drop all of its locally-stored information (such as megolm keys), and not just as an indication of whether things were dropped on the server-side.

Author: hmm, fair - will look into wording that a bit more safely.
